include/linux/file_compat.h

```diff
@@ -76,8 +76,17 @@ spl_filp_fallocate(struct file *fp, int mode, loff_t offset, loff_t len)
 #define	spl_filp_fsync(fp, sync)	vfs_fsync(fp, (fp)->f_dentry, sync)
 #endif /* HAVE_2ARGS_VFS_FSYNC */
 
+#ifdef HAVE_INODE_LOCK_SHARED
+#define	spl_inode_lock(ip)		inode_lock_nested(ip, I_MUTEX_PARENT)
+#define	spl_inode_unlock(ip)		inode_unlock(ip)
+#define	spl_inode_lock_shared(ip)	inode_lock_shared(ip)
+#define	spl_inode_unlock_shared(ip)	inode_unlock_shared(ip)
+#else
 #define	spl_inode_lock(ip)		mutex_lock_nested(&(ip)->i_mutex, I_MUTEX_PARENT)
 #define	spl_inode_unlock(ip)		mutex_unlock(&(ip)->i_mutex)
+#define	spl_inode_lock_shared(ip)	mutex_lock(&(ip)->i_mutex)
```

Comment:
It's an interesting thing that there's no `inode_lock_shared_nested`. Perhaps, they haven't found out a need to nest a shared lock. So for consistency, I'd still want to keep `spl_inode_lock_shared` non-nested.

Do you think I should map `spl_inode_lock` to non-nested and change the required users to spl_inode_lock_nested?